templates/check-java.sh.erb

```diff
@@ -1,26 +1,58 @@
 #!/bin/sh
 # This file is managed by Puppet
 
-# This file should be identical to 'check-java.sh' in the Jira 7.0.4 standalone
+# This file should be identical to 'check-java.sh' in the Jira 8.15.0 standalone
 # tarball.
 
-_EXPECTED_JAVA_VERSION="8"
-
 #
 # check for correct java version by parsing out put of java -version
-# we expect first line to be in format 'java version "1.8.0_40"' and assert that minor version number will be 8 or higher
+# we expect first line to be in format 'java version "1.8.0_161"' or 'java version "10.0.1" 2018-04-17'
+# or 'openjdk version "11-ea" 2018-09-25' and assert that version number will be 8 or 11 (if enabled)
+# or sth like 'Picked up JDK_JAVA_OPTIONS:' (which we need to skip)
 #
 
-"$_RUNJAVA" -version 2>&1 | grep "java version" | (
-        IFS=. read ignore1 version ignore2
-        if [ ! ${version:-0} -ge "$_EXPECTED_JAVA_VERSION" ]
-        then
-           echo "*************************************************************************************************************************************"
-           echo "**********     Wrong JVM version! You are running with "$ignore1"."$version"."$ignore2" but JIRA requires at least 1.8 to run.      **********"
-           echo "*************************************************************************************************************************************"
-           exit 1
-        fi
-    )
-if [ $? -ne 0 ] ; then
-   exit 1
+java_raw_version=`echo "$($_RUNJAVA -version 2>&1)" | grep -v "JDK_JAVA_OPTIONS" | grep "version" | awk '{ print substr($3, 2, length($3)-2); }'`
+java_version=0
+
+if [[ $java_raw_version = *-ea* ]]
```

Comment:
I guess this is upstream stuff, but these are bashisms.

Reply:
Yes, it is exactly upstream stuff (except the comment at the top).
Even if resolving bashisms and shellcheck problems would improve the script here, it might be a good idea to stick to upstream. Shouldn't we?

Reply:
Well, if we do that, then let's take the full path and avoid staying strongly in line with upstream.
I will update the PR with removed bashisms and a "green" shellcheck.

Also bash as shebang is no problem, since the script is called by catalina.sh, which itself has bash as shebang.

Reply:
I think I saw issues with it on Ubuntu where `/bin/sh` is `dash`, not `bash`. Perhaps dash on 18.04 does understand `[[` where 16.04 didn't.

Reply:
Updated the shebang now to /bin/bash.